pkg/wasm/main.go

```diff
@@ -0,0 +1,212 @@
+
+		return string(marshaled)
+	}))
+	fmt.Println("zed initialized")
+	<-c
+}
+
+func buildRootCmd() *cobra.Command {
+	rootCmd := &cobra.Command{
+		Use:   "zed",
+		Short: "The Authzed CLI",
+		Long:  "A client for managing Authzed from your command line.",
+	}
+
+	versionCmd := &cobra.Command{
+		Use:   "version",
+		Short: "display zed version information",
+		RunE:  versionCmdFunc,
+	}
+	cobrautil.RegisterVersionFlags(versionCmd.Flags())
+	rootCmd.AddCommand(versionCmd)
+
+	// Register shared commands.
+	commands.RegisterPermissionCmd(rootCmd)
+	commands.RegisterRelationshipCmd(rootCmd)
+	commands.RegisterSchemaCmd(rootCmd)
+	return rootCmd
+}
+
+func versionCmdFunc(cmd *cobra.Command, args []string) error {
+	console.Println(cobrautil.UsageVersion("zed", cobrautil.MustGetBool(cmd, "include-deps")))
+	return nil
+}
+
+func runZedCommand(rootCmd *cobra.Command, requestContextJSON string, stringParams []string) zedCommandResult {
+	ctx := context.Background()
+
+	// Decode the request context.
+	requestCtx := &devinterface.RequestContext{}
+	err := protojson.Unmarshal([]byte(requestContextJSON), requestCtx)
+	if err != nil {
+		return zedCommandResult{Error: err.Error()}
+	}
+
+	// Build the new dev context.
+	devCtx, devErrs, err := development.NewDevContext(ctx, requestCtx)
+	if err != nil {
+		return zedCommandResult{Error: err.Error()}
+	}
+	if devErrs != nil {
+		return zedCommandResult{Error: "invalid schema or relationships"}
+	}
+
+	// Run the V1 API against the dev context.
+	conn, stop, err := devCtx.RunV1InMemoryService()
+	defer stop()
+	if err != nil {
+		return zedCommandResult{Error: err.Error()}
+	}
+
+	// Set a NewClient function which constructs a wasmClient which passes
+	// all API calls to the in-memory V1 connection.
+	client.NewClient = func(cmd *cobra.Command) (client.Client, error) {
+		return wasmClient{
+			conn: conn,
+		}, nil
+	}
+
+	// Override Printf and Logger to collect the output.
+	var buf bytes.Buffer
+	console.Printf = func(format string, a ...any) {
+		fmt.Fprintf(&buf, format, a...)
+	}
+
+	log.Logger = zerolog.New(&buf).With().Bool("is-log", true).Timestamp().Logger()
+
+	// Set the input arguments.
+	rootCmd.SetArgs(stringParams)
+	rootCmd.SetOut(&buf)
+	rootCmd.SetErr(&buf)
+
+	// Execute the command.
+	err = rootCmd.Execute()
+	if err != nil {
+		return zedCommandResult{Error: err.Error()}
+	}
+
+	// Collect the updated schema and relationships.
+	headRev, err := devCtx.Datastore.HeadRevision(ctx)
+	if err != nil {
+		return zedCommandResult{Error: err.Error()}
+	}
+
+	reader := devCtx.Datastore.SnapshotReader(headRev)
+	relationships := []*core.RelationTuple{}
+
+	nsDefs, err := reader.ListNamespaces(ctx)
+	if err != nil {
+		return zedCommandResult{Error: err.Error()}
+	}
+
+	for _, nsDef := range nsDefs {
+		it, err := reader.QueryRelationships(ctx, datastore.RelationshipsFilter{ResourceType: nsDef.Name})
+		if err != nil {
+			return zedCommandResult{Error: err.Error()}
+		}
+		defer it.Close()
```

Comment:
No need for this defer

Reply:
I always put the `defer it.Close()`, even if I close it after the loop, to *ensure* it will be closed even if the code is changed in the future